soccer/Processor.cpp

```diff
@@ -582,6 +608,59 @@ void Processor::run() {
     vision.stop();
 }
 
+void Processor::decodeGeometryPacket(const SSL_GeometryFieldSize* fieldSize) {
+    if (fieldSize->field_lines_size() == 0) {
+        return;
+    }
+
+    const SSL_FieldCicularArc* penalty = nullptr;
+    const SSL_FieldCicularArc* center = nullptr;
+    float displacement = 0.500f;  // default displacment
+
+    for (int i = 0; i < fieldSize->field_arcs().size(); i++) {
+        if (boost::iequals(fieldSize->field_arcs().Get(i).name(), "CenterCircle")) {
+            // Assume center circle
+            center = &fieldSize->field_arcs().Get(i);
+        } else if (boost::iequals(fieldSize->field_arcs().Get(i).name(), "LeftFieldLeftPenaltyArc")) {
+            penalty = &fieldSize->field_arcs().Get(i);
+        }
+    }
+
+    for (int i = 0; i < fieldSize->field_lines().size(); i++) {
+        if (boost::iequals(fieldSize->field_lines().Get(i).name(), "RightPenaltyStretch")) {
+            displacement = abs(fieldSize->field_lines().Get(i).p2().y() - fieldSize->field_lines().Get(i).p1().y());
+        }
+    }
+
+    float thickness = fieldSize->field_lines().Get(0).thickness() / 1000.0f;
+
+    // The values we get are the center of the lines, we want to use the outside, so we can add this as an offset.
+    float adj = fieldSize->field_lines().Get(0).thickness() / 1000.0f / 2.0f;
+
+    float fieldBorder = currentDimensions->Border();
+
+    if (penalty != nullptr && center != nullptr && thickness != 0) {
+        // Force a resize
+        // TODO fix hardcoded values here
+        setFieldDimensions(Field_Dimensions (
+                               fieldSize->field_length() / 1000.0f, fieldSize->field_width() / 1000.0f,
+                               fieldBorder,
+                               thickness,
+                               fieldSize->goal_width() / 1000.0f, fieldSize->goal_depth() / 1000.0f,
+                               0.160f,                                  // Goal Height
+                               penalty->radius() / 1000.0f + adj,       // PenaltyDist
+                               0.010f,                                  // PenaltyDiam
```

Comment:
For this and goal height, let's add `static const float Default___` values to the Field_Dimensions class so we don't have these constants both here and in `Field_Dimensions.cpp`.